src/chunk_processors.py

```diff
@@ -298,52 +275,244 @@ class CombineTextChunksIntoList(PipelineComponent):
     """
     Combines consecutive text chunks that match a list item pattern into list chunks.
 
+    Also incorporates preceding chunks that end with a colon as part of the list,
+    since these often introduce the list.
+
     If used in a pipeline with `CombineSuccessiveSameTypeChunks` on type TEXT, this
     should go before that.
 
-    TODO: handle cases where a list item is split across multiple text blocks
+    TODO: this will join multiple consecutive lists with introductions ending in colons.
     """
 
     def __init__(self, text_separator: str = "\n") -> None:
         self.text_separator = text_separator
         self.list_item_pattern = (
-            r"(^|\n)(?:•|-|(?:[\(|\[]?[0-9a-zA-Z]{0,3}[\.|\)|\]])).*?"
+            r"(^|\n)(?:•|-|·|(?:[\(|\[]?[0-9a-zA-Z]{0,3}[\.|\)|\]])).*?"
         )
 
     def __call__(self, chunks: list[Chunk]) -> list[Chunk]:
         """Run list item combining."""
         new_chunks: list[Chunk] = []
         current_list_chunk = None
+        potential_list_continuation = False
+        potential_list_intro = None
```

Comment:
this `list_intro` stuff is a v smart addition 🤓

v interested in what you find in asking/answering this question! 👇
>Although I'm not sure whether those list 'introductions' could be headings instead and lists could be separated into list items, this is left for after I have a look at this with Anne

Reply:
i'm gonna leave it to after MVP and focus on deployment for now i think - too much of an interesting-looking rabbit hole for now 😇